testing/cli/test_cli_sct_analyze_lesion.py

```diff
@@ -89,3 +90,37 @@ def test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements(dummy_les
                 assert measurements.at[0, key] < expected_value
             elif key == 'length [mm]':
                 assert measurements.at[0, key] > expected_value
+
+
+@pytest.mark.sct_testing
+@pytest.mark.parametrize("dummy_lesion, rtol", [
+    # Straight region of `t2.nii.gz` -> little curvature -> smaller tolerance
+    ([(29, 45, 25), (3, 10, 2)], 0.001),
+    ([(29, 27, 25), (1, 4, 1)], 0.001),  # NB: Example from #3633
+    # Curved region of `t2.nii.gz` -> lots of curvature -> larger tolerance
+    ([(29, 0, 25), (4, 15, 3)], 0.01)
+], indirect=["dummy_lesion"])
+def test_sct_analyze_lesion_matches_expected_dummy_lesion_measurements_without_segmentation(dummy_lesion, rtol,
+                                                                                            tmp_path):
+    """Run the CLI script without providing SC segmentation -- only volume is computed. Max_equivalent_diameter and
+    length are nan."""
+    # Run the analysis on the dummy lesion file
+    path_lesion, expected_measurements = dummy_lesion
+    sct_analyze_lesion.main(argv=['-m', path_lesion,
+                                  '-ofolder', str(tmp_path)])
+
+    # Load analysis results from pickled pandas.Dataframe
+    _, fname, _ = extract_fname(path_lesion)
+    with open(tmp_path/f"{fname}_analysis.pkl", 'rb') as f:
+        measurements = pickle.load(f)['measures']
+
+    # Validate analysis results
+    for key, expected_value in expected_measurements.items():
+        if key == 'volume [mm3]':
+            np.testing.assert_equal(measurements.at[0, key], expected_value)
+        else:
+            # The max_equivalent_diameter and length are nan because no segmentation is provided
+            if key == 'max_equivalent_diameter [mm]':
+                assert math.isnan(measurements.at[0, key]) is True
+            elif key == 'length [mm]':
+                assert math.isnan(measurements.at[0, key]) is True
```

Comment:
For normal code, I would agree that such duplication means we should look for a common abstraction, but for testing code I'm not so sure. Normally, I would advocate for the most straightforward code possible in tests, and in particular, as little control flow as possible (ifs, loops, etc.). For testing code, if that means a small amount of copy-pasted code, I think that's fine.

Here, you're right that `args` could be parametrized easily, but I don't think the diameter/length can be parametrized straightforwardly, since it involves `nan` comparisons. (It might be possible to do something tricky with `nan` comparisons, like `assert not X >= expected`, but I don't think that would be nice to read and reason about when debugging a failing test.)

On a different note, though, I will point out that assertions of the form:
```py3
assert X is True
assert Y is False
```
can be better written as the shorter:
```py3
assert X
assert not Y
```

Reply:
I agree with @mguaypaq. Even the [first parametrization](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/86c96028ad112fad614596cb22df6e527978793e/testing/cli/test_cli_sct_analyze_lesion.py#L96-L102) had to be explained to me by @mguaypaq. So, I would vote for keeping the tests split to be understandable even for less experienced users (like me 😄). 

Regarding the`assert` simplification:

> On a different note, though, I will point out that assertions of the form:
> 
> ```python
> assert X is True
> assert Y is False
> ```
> 
> can be better written as the shorter:
> 
> ```python
> assert X
> assert not Y
> ```

I have done this in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4112/commits/86c96028ad112fad614596cb22df6e527978793e, together with simplification of `if-else` logic.

Reply:
I am okay with this plan. :)

> but I don't think the diameter/length can be parametrized straightforwardly, since it involves `nan` comparisons.

Just to clarify my thought process: rather than checking against `nan`, I was going to propose not checking length/diameter at all for the no-seg case. So, the changes to the `if` statements would be rather simple, e.g.:

```diff
-            if key == 'max_equivalent_diameter [mm]':
+            if path_seg and key == 'max_equivalent_diameter [mm]':
```

I was thinking that because the user isn't expecting length/diameter if they're not providing a seg, then they don't really care whether it's nan? But, maybe there's more value in ensuring that these values are nan than I'm giving credit for.

Reply:
Anyway, I think this PR is good to go now?

Reply:
> Anyway, I think this PR is good to go now?

Agree --> merging